pkgs/build-support/rust/lib/default.nix

```
@@ -12,17 +12,25 @@ rec {
   # hostPlatform-targeted compiler -- for example, `-m64` being
   # passed on a build=x86_64/host=aarch64 compilation.
   envVars = let
+
+    isAarch64Static = platform: platform.isAarch64 && platform.isMusl && platform.isStatic;
```

It's _probably_ not musl-specific, right?  The issue is just static libraries not having dependency information, as I understand it.  But we wouldn't want to link with lld on Darwin, so maybe

```suggestion
    shouldUseLLD = platform: platform.isAarch64 && platform.isStatic && !platform.isDarwin;
```

I also think we need some explanation of why we're using lld, I guess with a link to the issue.